### PR TITLE
add CRD trainingjob status definition

### PR DESCRIPTION
**Pull request by qizheng09**
Hey, @typhoonzero @Yancey1989 
Let's talk about the status definition of trainingjob and do you have any advice about my proposal ?

### CODE REVIEW

File: go/pkg/apis/paddlepaddle/v1alpha1/types.go [hunk 1]
```diff
@@ -73,9 +73,63 @@ type TrainerSpec struct {
...
+	ResourceStateStarting                = "starting"
+	ResourceStateRunning                 = "running"
+	ResourceStateFailed                  = "failed"
+	ResourceStateSucceeded               = "succeeded"
+)
+
+// Detail status of every resource state
+type TrainingResourceStatus struct {
+	// TrainingResourceType the type of TrainingJob resource, include MASTER PSERVER and TRAINER
+	TrainingResourceType `json:"pd_peplica_type"`
```
Comment (m3ngyang) [inline go/pkg/apis/paddlepaddle/v1alpha1/types.go]: `pd_peplica_type` typo

File: go/pkg/apis/paddlepaddle/v1alpha1/types.go [hunk 2]
```diff
@@ -73,9 +73,63 @@ type TrainerSpec struct {
...
+const (
+	TrainingJobJobPhaseNone      TrainingJobPhase = ""
+	TrainingJobJobPhaseCreating                   = "creating"
+	TrainingJobJobPhaseRunning                    = "running"
+	TrainingJobJobPhaseSucceeded                  = "succeeded"
+	TrainingJobJobPhaseFailed                     = "failed"
+)
+
+// The scale status of trainer jobs
+type TrainingScaleStatus struct {
```
Comment (m3ngyang) [inline go/pkg/apis/paddlepaddle/v1alpha1/types.go]: Commit should start with "TrainingScaleStatus ..."

Comment (typhoonzero) [reply go/pkg/apis/paddlepaddle/v1alpha1/types.go]: Since it's the struct for trainer jobs, the name should be `TrainerJobScaleStatus`?

**Comment by qizheng09** [comment]
@typhoonzero Thank you for your comments. I have fixed some typos and some annotations.

**Pull request merged**